### PR TITLE
feat(rome_cli): cli exit as error when no files are found

### PR DESCRIPTION
**Pull request by xunilrj**
## Summary

Closes https://github.com/rome/tools/issues/3645

This PR makes the cli exit as error if no files are found.

## Test Plan

```
> cd c:\Python310
> rome check .
Checked 0 files in 79ms
Error: no supported files were found in the specified folders.
> $LASTEXITCODE
1
```

### CODE REVIEW

File: crates/rome_cli/src/termination.rs [hunk 1]
```diff
@@ -76,6 +76,9 @@ pub enum Termination {
 
     #[error("The combination of configuration and arguments is invalid: \n {0}")]
     IncompatibleEndConfiguration(&'static str),
+
+    #[error("no supported files were found in the specified folders.")]
+    ZeroSupportedFilesFound,
```
Comment (ematipico) [inline crates/rome_cli/src/termination.rs:81]: nit: `NoSupportedFilesFound` ?

**Review by ematipico — CHANGES_REQUESTED** [review]
Could you please add a new test case so we can snapshot the error message?

**Review by ematipico — APPROVED** [review]

File: crates/rome_cli/Cargo.toml [hunk 2]
```diff
@@ -49,3 +49,4 @@ tikv-jemallocator = "0.5.0"
 [dev-dependencies]
 insta = "1.18.2"
 tokio = { workspace = true, features = ["io-util"] }
+mktemp = "0.5.0"
```
Comment (MichaReiser) [inline crates/rome_cli/Cargo.toml:52]: What is the `mktemp` dependency used for?

File: crates/rome_cli/tests/snapshots/main_commands_check/no_supported_file_found.snap [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+source: crates/rome_cli/tests/snap_test.rs
+expression: content
+---
+# Termination Message
+
+```block
+no supported files were found in the specified folders.
```
Comment (MichaReiser) [inline crates/rome_cli/tests/snapshots/main_commands_check/no_supported_file_found.snap:8]: Nit: 
```suggestion
no supported files were found in the specified paths.
```

File: crates/rome_cli/src/termination.rs (hunk 1 above)
Comment (xunilrj) [reply crates/rome_cli/src/termination.rs]: Done. Actually went with `NoFilesWereProcessed`.

**Comment by xunilrj** [comment]
@leops I changed the semantic of the processed variable. Are you ok with this change?

**Comment by xunilrj** [comment]
> Could you please add a new test case so we can snapshot the error message?

Done.

File: crates/rome_cli/Cargo.toml (hunk 2 above)
Comment (xunilrj) [reply crates/rome_cli/Cargo.toml:52]: Removed.

File: crates/rome_cli/tests/snapshots/main_commands_check/no_supported_file_found.snap (hunk 3 above)
Comment (xunilrj) [reply crates/rome_cli/tests/snapshots/main_commands_check/no_supported_file_found.snap]: Done.

**Review by ematipico — APPROVED** [review]

**Pull request merged**